RoddyCore/src/de/dkfz/roddy/core/Analysis.groovy

```diff
@@ -32,142 +32,145 @@ import static de.dkfz.roddy.tools.RoddyIOHelperMethods.getStackTraceAsString
+    Analysis(String name, Project project, RuntimeService runtimeService, AnalysisConfiguration configuration) {
+        this.name = name
+        this.project = project
+        this.configuration = configuration
+        this.runtimeService = runtimeService
     }
 
-    public String getName() {
-        return name;
+    String getName() {
+        return name
     }
 
-    public Project getProject() {
-        return project;
+    Project getProject() {
+        return project
     }
 
-    public String getUsername() {
+    String getUsername() {
         try {
-            return FileSystemAccessProvider.getInstance().callWhoAmI();
+            return FileSystemAccessProvider.getInstance().callWhoAmI()
         } catch (Exception e) {
-            return "UNKNOWN";
+            return "UNKNOWN"
         }
     }
 
-    public String getUsergroup() {
+    String getUsergroup() {
         try {
             //Get the default value.
-            String groupID = FileSystemAccessProvider.getInstance().getMyGroup();
+            String groupID = FileSystemAccessProvider.getInstance().getMyGroup()
 
             //If it is configured, get the group id from the config.
-            boolean processSetUserGroup = getConfiguration().getConfigurationValues().getBoolean(ConfigurationConstants.CVALUE_PROCESS_OPTIONS_SETUSERGROUP, true);
+            boolean processSetUserGroup = getConfiguration().getConfigurationValues().getBoolean(ConfigurationConstants.CVALUE_PROCESS_OPTIONS_SETUSERGROUP, true)
             if (processSetUserGroup) {
-                groupID = getConfiguration().getConfigurationValues().getString(ConfigurationConstants.CFG_OUTPUT_FILE_GROUP, groupID);
+                groupID = getConfiguration().getConfigurationValues().getString(ConfigurationConstants.CFG_OUTPUT_FILE_GROUP, groupID)
             }
-            return groupID;
+            return groupID
         } catch (Exception e) {
-            return "UNKNOWN";
+            return "UNKNOWN"
         }
     }
 
-    /**
-     * The object should actually move to context itself. However, currently, this would break a lot.
-     * Mark it as Deprecated for now.
-     */
-    @Deprecated
-    private ContextConfiguration _contextConfiguration = null;
+    private ContextConfiguration _analysisConfiguration = null
 
-    public AnalysisConfiguration getConfiguration() {
-        if (_contextConfiguration == null)
-            _contextConfiguration = new ContextConfiguration((AnalysisConfiguration) this.configuration, (ProjectConfiguration) this.project.getConfiguration());
-        return _contextConfiguration;
-//        return configuration;
+    AnalysisConfiguration getConfiguration() {
+        if (_analysisConfiguration == null) {
+            _analysisConfiguration = new ContextConfiguration((AnalysisConfiguration) this.configuration, (ProjectConfiguration) this.project.getConfiguration())
+            [
+                    'USERNAME' : getUsername(),
```

Comment:
This is a repetition of the identifiers "USERNAME" etc. Better put them into consants.